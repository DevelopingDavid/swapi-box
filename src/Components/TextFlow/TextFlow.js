import React from 'react';

const TextFlow = (props) => {
  return (
    <section className="textFlow-container">
      <div className="fade"></div>
      <div className="star-wars">
        <div className="crawl">
          <div className="title">
            <h3>{props.movieInfo.releaseDate}</h3>
            <h1>{props.movieInfo.movieTitle}</h1>
          </div>
          <p>{props.movieInfo.crawlText}</p>
        </div>
      </div>
    </section>
  )
}

export default TextFlow;