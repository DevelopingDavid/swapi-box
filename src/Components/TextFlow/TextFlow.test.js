import React from 'react';
import { shallow } from 'enzyme';
import TextFlow from './TextFlow';

const mockState = {crawlText: 'It is a period of civil war.\r\nRebel spaceships', movieTitle: 'A New Hope', releaseDate: '1977-05-25'}

describe('TextFlow', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <TextFlow movieInfo={mockState}/>
    )
  });

  it('should create snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
});