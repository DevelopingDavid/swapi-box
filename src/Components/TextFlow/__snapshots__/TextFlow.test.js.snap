// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextFlow should create snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextFlow
    movieInfo={
      Object {
        "crawlText": "It is a period of civil war.
Rebel spaceships",
        "movieTitle": "A New Hope",
        "releaseDate": "1977-05-25",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="fade"
        />,
        <div
          className="star-wars"
        >
          <div
            className="crawl"
          >
            <div
              className="title"
            >
              <h3>
                1977-05-25
              </h3>
              <h1>
                A New Hope
              </h1>
            </div>
            <p>
              It is a period of civil war.
Rebel spaceships
            </p>
          </div>
        </div>,
      ],
      "className": "textFlow-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "fade",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="crawl"
          >
            <div
              className="title"
            >
              <h3>
                1977-05-25
              </h3>
              <h1>
                A New Hope
              </h1>
            </div>
            <p>
              It is a period of civil war.
Rebel spaceships
            </p>
          </div>,
          "className": "star-wars",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title"
              >
                <h3>
                  1977-05-25
                </h3>
                <h1>
                  A New Hope
                </h1>
              </div>,
              <p>
                It is a period of civil war.
Rebel spaceships
              </p>,
            ],
            "className": "crawl",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3>
                    1977-05-25
                  </h3>,
                  <h1>
                    A New Hope
                  </h1>,
                ],
                "className": "title",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "1977-05-25",
                  },
                  "ref": null,
                  "rendered": "1977-05-25",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "A New Hope",
                  },
                  "ref": null,
                  "rendered": "A New Hope",
                  "type": "h1",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "It is a period of civil war.
Rebel spaceships",
              },
              "ref": null,
              "rendered": "It is a period of civil war.
Rebel spaceships",
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="fade"
          />,
          <div
            className="star-wars"
          >
            <div
              className="crawl"
            >
              <div
                className="title"
              >
                <h3>
                  1977-05-25
                </h3>
                <h1>
                  A New Hope
                </h1>
              </div>
              <p>
                It is a period of civil war.
Rebel spaceships
              </p>
            </div>
          </div>,
        ],
        "className": "textFlow-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fade",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="crawl"
            >
              <div
                className="title"
              >
                <h3>
                  1977-05-25
                </h3>
                <h1>
                  A New Hope
                </h1>
              </div>
              <p>
                It is a period of civil war.
Rebel spaceships
              </p>
            </div>,
            "className": "star-wars",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="title"
                >
                  <h3>
                    1977-05-25
                  </h3>
                  <h1>
                    A New Hope
                  </h1>
                </div>,
                <p>
                  It is a period of civil war.
Rebel spaceships
                </p>,
              ],
              "className": "crawl",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h3>
                      1977-05-25
                    </h3>,
                    <h1>
                      A New Hope
                    </h1>,
                  ],
                  "className": "title",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "1977-05-25",
                    },
                    "ref": null,
                    "rendered": "1977-05-25",
                    "type": "h3",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "A New Hope",
                    },
                    "ref": null,
                    "rendered": "A New Hope",
                    "type": "h1",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "It is a period of civil war.
Rebel spaceships",
                },
                "ref": null,
                "rendered": "It is a period of civil war.
Rebel spaceships",
                "type": "p",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
